### PR TITLE
feat: Add pagination support to console logs to prevent LLM token limits

### PR DESCRIPTION
**Pull request by Saqoosha**
## 🚨 Problem

When Unity console logs accumulate in large quantities (hundreds to thousands of entries), the `get_console_logs` function fails due to **LLM token limit exceeded** errors.

- **Actual Error**: `"response (27806 tokens) exceeds maximum allowed tokens (25000)"`
- **Impact**: MCP Unity debugging functionality becomes completely unusable
- **Occurrence**: During extended Unity development sessions

### **Pagination Implementation**
- Added `offset` and `limit` parameters for chunked log retrieval
- Newest-first chronological sorting for optimal debugging workflow
- Detailed pagination metadata (totalCount, hasMore, etc.)

### **Memory Management Enhancement**
- Automatic cleanup functionality (max 1000 entries, remove 200 at a time)
- Memory-efficient log management

### **Consistent Implementation Across Both Sides**
- **Unity Side (C#)**: Added `GetLogsAsJson(logType, offset, limit)` method
- **Node.js Side (TypeScript)**: Extended parameter handling and URL templates

### **Dramatic Token Usage Reduction**
- **Before**: 27,806 tokens → Error
- **After**: ~1,000 tokens (**96% reduction**)

### **Recommended Usage Examples**
```typescript
// Check latest 5 errors
get_console_logs({ limit: 5 })

// Retrieve specific page
get_console_logs({ offset: 10, limit: 20 })

// Paginated error log retrieval
get_console_logs({ logType: "error", limit: 10 })
```

### **Performance Improvements**
- Dramatically reduced response times
- Stable operation in large log environments
- Significantly improved debugging efficiency

## 🧪 Testing

Thoroughly tested with large volumes of test logs (Error/Exception/Assert/Warning/Info):
- Pagination functionality accuracy
- Stability under boundary conditions
- Memory management effectiveness

### **Unity Side Changes**
- `Editor/Services/ConsoleLogsService.cs`: Added paginated `GetLogsAsJson` method
- `Editor/Services/IConsoleLogsService.cs`: Extended interface with pagination support
- `Editor/Resources/GetConsoleLogsResource.cs`: Added parameter extraction and processing

### **Node.js Side Changes**
- `Server~/src/tools/getConsoleLogsTool.ts`: Added offset/limit parameters to Zod schema
- `Server~/src/resources/getConsoleLogsResource.ts`: Extended URL template to support pagination

### **Key Features**
- **Newest-first ordering**: Most recent logs appear first for optimal debugging
- **Automatic memory cleanup**: Prevents memory bloat during long sessions
- **Detailed pagination info**: Provides comprehensive metadata for navigation
- **Backward compatibility**: Existing `get_console_logs` calls continue to work unchanged

**Breaking Changes**: None (fully backward compatible)

### CODE REVIEW

File: Editor/Resources/GetConsoleLogsResource.cs [hunk 1]
```diff
@@ -13,38 +13,52 @@ public class GetConsoleLogsResource : McpResourceBase
...
+            JObject result = _consoleLogsService.GetLogsAsJson(logType, offset, limit);
+            
+            // Add success info to the response
+            result["success"] = true;
+            
+            var pagination = result["pagination"] as JObject;
+            string typeFilter = logType != null ? $" of type '{logType}'" : "";
+            int returnedCount = pagination?["returnedCount"]?.Value<int>() ?? 0;
+            int filteredCount = pagination?["filteredCount"]?.Value<int>() ?? 0;
+            result["message"] = $"Retrieved {returnedCount} of {filteredCount} log entries{typeFilter} (offset: {offset}, limit: {limit})";
```
Comment (CoderGamester) [inline Editor/Resources/GetConsoleLogsResource.cs:45]: change the return to be the same JObject format as before
is a more readable and direct text for the AI agent to process

File: Editor/Services/ConsoleLogsService.cs [hunk 2]
```diff
@@ -79,10 +83,26 @@ public void StopListening()
         /// </summary>
         /// <returns>JArray containing all logs</returns>
         public JArray GetAllLogsAsJson(string logType = "")
```
Comment (CoderGamester) [inline Editor/Services/ConsoleLogsService.cs:85]: remove old method

File: Editor/Services/ConsoleLogsService.cs [hunk 3]
```diff
@@ -101,21 +121,52 @@ public JArray GetAllLogsAsJson(string logType = "")
             
             lock (_logEntries)
             {
+                // First pass: count total and filtered entries
                 foreach (var entry in _logEntries)
```
Comment (CoderGamester) [inline Editor/Services/ConsoleLogsService.cs:125]: change to one single for loop for efficiency
if there is a limit and offset, that can be counted in the for loop variables

File: Editor/Services/ConsoleLogsService.cs [hunk 4]
```diff
@@ -101,21 +121,52 @@ public JArray GetAllLogsAsJson(string logType = "")
...
+                    currentIndex++;
+                    if (logsArray.Count >= limit) break;
                 }
             }
             
-            return logsArray;
+            return new JObject
+            {
+                ["logs"] = logsArray,
+                ["pagination"] = new JObject
```
Comment (CoderGamester) [inline Editor/Services/ConsoleLogsService.cs:160]: pagination is unecessary
the message log above is already complete to be fullfiled with the total logs count

File: Server~/src/resources/getConsoleLogsResource.ts [hunk 5]
```diff
@@ -19,25 +19,25 @@ function listLogTypes(resourceMimeType: string) {
       {
         uri: `unity://logs/`,
```
Comment (CoderGamester) [inline Server~/src/resources/getConsoleLogsResource.ts:20]: for the resource update the template with default offset and limit values
use max values that can be accepted

do this to all template commands

**Review by CoderGamester — CHANGES_REQUESTED** [review]
Very good PR suggestion

I made some comments that require changes to properly work

**Review by CoderGamester — APPROVED** [review]
LGTM

**Comment by CoderGamester** [comment]
@Saqoosha thank you very much for the PR
